libusb/os/windows_winusb.c

```diff
@@ -1387,9 +1388,14 @@ static int set_composite_interface(struct libusb_context *ctx, struct libusb_dev
 	// devices will have only MI_00 & MI_03 for instance), we retrieve the actual
 	// interface number from the path's MI value
 	mi_str = strstr(device_id, "MI_");
-	if ((mi_str != NULL) && isdigit((unsigned char)mi_str[3]) && isdigit((unsigned char)mi_str[4])) {
-		interface_number = ((mi_str[3] - '0') * 10) + (mi_str[4] - '0');
-	} else {
+        // Initialize to default
+        interface_number = 0;
```

Comment:
Don't think you need to initialize this one, because there should always be a MI_xx field, right?

Reply:
Unfortunately, MSVC appears to be not smart enough to figure out that the following will initialize the variable one way or another:
```C
	if (mi_str != NULL) {
		interface_number = strtoul(&mi_str[3], &endptr, 16);
	}

        if (mi_str == NULL || endptr - &mi_str[3] != 2) {
		usbi_warn(ctx, "failure to read interface number for %s, using default value", device_id);
		interface_number = 0;
	}
```
It throws _uninitialized variable_ warning treated as error

Reply:
Okay :) Same for endptr I guess.